src/fenwick.jl

```diff
@@ -0,0 +1,96 @@
+"""
+    FenwickTree{T}(n)
+    
+Constructs a [`FenwickTree`](https://en.wikipedia.org/wiki/Fenwick_tree) of length `n`.
+ 
+"""
+mutable struct FenwickTree{T}
+    bit::Vector{T}
```

Comment:
I found that I was confused by this name being used everywhere.  Googling around, it seems this might be an abbreviation of "binary index tree" (which is another name for a Fenwick tree).

In this package, most other data structures use more verbose names, I would suggest renaming this `binary_index_tree`, or possibly `bi_tree` (but add a comment describing what it means).

`store` is another option (e.g., as used by the `Deque` data structure).

It is true that other data structures here sometimes use short names--e.g., [here](https://github.com/JuliaCollections/DataStructures.jl/blob/69c867fcccc8a8210b4d841ba0ea0b4e9487a660/src/sorted_dict.jl#L5).  
But the type is also specified right after that, making it easy to understand the acronym.

The term, `bit`, though, is also a common computing term, and when I was looking at it, all I could think of is, what do "bits" have to do with Fenwick trees (other than the fact that it's implemented on a computer).